MCprep_addon/world_tools.py

```diff
@@ -973,6 +974,126 @@ def world_time_update(self, context):
 	return
 
 
+class MCPREP_OT_render_panorama(bpy.types.Operator):
```

Comment:
Probably the most helpful thing here is to switch this to also inherent from the `ImportHelper` class blender provides. This will change the invoke experience to be directly a file selection operation in a modal window. There is still a sidebar for additional properties in this modal, but most users probably won't see those so I wouldn't depend on it (but we can start there).

See how this is utilized with the swap texture pack button, same concept:
https://github.com/TheDuckCow/MCprep/blob/55719ce2b1faa3d007f6b2c69c7d0083ac26a90a/MCprep_addon/materials/prep.py#L342

(ignore the other class inherence McprepMaterialProps in this above example)

Reply:
I feel stupid for not finding this, I searched for forever... Thank you, that pretty much fixes it. I'm a bit unhappy with the sidebar, but people will just render 1024 I guess, unless they know what they are looking for anyways